### PR TITLE
Move common test components into FunderMaps.Testing

### LINKED ISSUE #232: Group test helpers in one common test base

### PR DESCRIPTION
**Pull request by yorickdewid**
- Move Repositories to FunderMaps.Testing
- Move Fakers to FunderMaps.Testing
- Move some helpers into FunderMaps.Testing
- Move some extensions into FunderMaps.Testing
- Remove ObjectStore
- Replace EntityDataStore with generic DataStore
- Add FunderMaps.Testing to solution

**What issue does this PR address?**
Fixes #232

**Does this PR introduce a breaking change?**
Yes. This breaks the package upgrade in #267 

**Other information**:

### CODE REVIEW

File: tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationAdminTests.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public async Task GetOrganizationByIdReturnSingleOrganization()
             // Arrange
             var organization = new OrganizationFaker().Generate();
             var client = _factory
-                .WithAuthentication(options => options.User.Role = ApplicationRole.Administrator)
```
Comment (tabeckers) [inline tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationAdminTests.cs:28]: Maybe move client creation to the constructor?

File: tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationProposalTests.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public async Task CreateOrganizationProposalReturnOrganizationProposal()
             // Arrange
             var organization = new OrganizationProposalDtoFaker().Generate();
             var client = _factory
-                .WithAuthentication(options => options.User.Role = ApplicationRole.Administrator)
```
Comment (tabeckers) [inline tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationProposalTests.cs:28]: Maybe move client creation to the constructor?

File: tests/FunderMaps.Testing/Faker/UserFaker.cs [hunk 3]
```diff
@@ -104,6 +107,9 @@ public class UserFaker : Faker<User>
             "writer",
         };
 
+        /// <summary>
+        ///     Creatw new instance.
```
Comment (tabeckers) [inline tests/FunderMaps.Testing/Faker/UserFaker.cs:111]: It's a typo wow

Comment (yorickdewid) [reply tests/FunderMaps.Testing/Faker/UserFaker.cs]: Fixed

File: tests/FunderMaps.IntegrationTests/CustomWebApplicationFactory.cs [hunk 4]
```diff
@@ -48,44 +48,36 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
             builder.ConfigureTestServices(ConfigureTestServices);
         }
 
-        // FUTURE Helper, move ?
-        public virtual CustomWebApplicationFactory<TStartup> WithDataStoreList<TEntity>(TEntity entity)
-            where TEntity : BaseEntity<TEntity>
+        public virtual CustomWebApplicationFactory<TStartup> WithDataStoreList<TItem>(TItem item)
```
Comment (tabeckers) [inline tests/FunderMaps.IntegrationTests/CustomWebApplicationFactory.cs:51]: Shouldn't this be `WithDataStoreItem`?

File: tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationAdminTests.cs (hunk 1 above)
Comment (yorickdewid) [reply tests/FunderMaps.IntegrationTests/Backend/Application/OrganizationAdminTests.cs:28]: Complicates things in many cases, prefer not to unless an entire test collection operates on the same client

File: tests/FunderMaps.Testing/Repositories/TestAnalysisRepository.cs [hunk 5]
```diff
@@ -9,35 +9,45 @@
...
+        public TestAnalysisRepository(DataStore<AnalysisProduct> dataStore)
+        {
+            DataStore = dataStore;
+        }
 
-        public Task<IEnumerable<AnalysisProduct>> GetAllInFenceAsync(Guid userId, INavigation navigation, CancellationToken token) => throw new NotImplementedException();
+        public Task<IEnumerable<AnalysisProduct>> GetAllInFenceAsync(Guid userId, INavigation navigation, CancellationToken token)
+            => throw new NotImplementedException();
 
         public async Task<AnalysisProduct> GetByExternalIdAsync(Guid userId, string externalId, ExternalDataSource externalSource, CancellationToken token)
```
Comment (tabeckers) [inline tests/FunderMaps.Testing/Repositories/TestAnalysisRepository.cs:35]: None of these should actually be async (you may assign this to me though)

File: tests/FunderMaps.Testing/Repositories/TestOrganizationUserRepository.cs [hunk 6]
```diff
@@ -7,32 +7,26 @@
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace FunderMaps.IntegrationTests.Repositories
+namespace FunderMaps.Testing.Repositories
 {
-    public class OrganizationUserRecord : BaseEntity<OrganizationUserRecord>
+    public class OrganizationUserRecord
```
Comment (tabeckers) [inline tests/FunderMaps.Testing/Repositories/TestOrganizationUserRepository.cs:12]: Separate file & folder?

File: tests/FunderMaps.Testing/Repositories/TestUserRepository.cs [hunk 7]
```diff
@@ -6,50 +6,43 @@
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace FunderMaps.IntegrationTests.Repositories
+namespace FunderMaps.Testing.Repositories
 {
-    public class UserRecord : BaseEntity<UserRecord>
+    public class UserRecord
```
Comment (tabeckers) [inline tests/FunderMaps.Testing/Repositories/TestUserRepository.cs:11]: Separate file & folder?

**Review by tabeckers — CHANGES_REQUESTED** [review]
Left comments in files where required. A bigger question is the creation of the client in the constructor instead of in each method itself. Should we do this?

File: tests/FunderMaps.IntegrationTests/CustomWebApplicationFactory.cs (hunk 4 above)
Comment (yorickdewid) [reply tests/FunderMaps.IntegrationTests/CustomWebApplicationFactory.cs:51]: Forgot about those

Comment (yorickdewid) [reply tests/FunderMaps.IntegrationTests/CustomWebApplicationFactory.cs]: Done

File: tests/FunderMaps.Testing/Repositories/TestAnalysisRepository.cs (hunk 5 above)
Comment (yorickdewid) [reply tests/FunderMaps.Testing/Repositories/TestAnalysisRepository.cs:35]: The 'easiest' solution is to make them await an empty task. The solution is sufficient. Returning an incomplete task is more reliable in most situations however performance is not a consideration when SUT.

File: tests/FunderMaps.Testing/Repositories/TestOrganizationUserRepository.cs (hunk 6 above)
Comment (yorickdewid) [reply tests/FunderMaps.Testing/Repositories/TestOrganizationUserRepository.cs:12]: At best it be a internal entity/type. There may no need for a separate unit

File: tests/FunderMaps.Testing/Repositories/TestUserRepository.cs (hunk 7 above)
Comment (yorickdewid) [reply tests/FunderMaps.Testing/Repositories/TestUserRepository.cs:11]: Idem

**Review by tabeckers — APPROVED** [review]
Looks good

**Pull request merged**